### DISCUSSION TITLE
Dirac Kernels for INSAD

### DISCUSSION THREAD

**Discussion by maxnezdyur** [idaholab/moose#23378] [Q&A Modules: Navier-Stokes] [answered]
Is there a simple way to have DiracKernels work with Vector Variables? Not sure if there is a way to apply it component-wise. Thanks!

**Comment by lindsayad** [comment, marked as answer]
I think the answer is no. This is the first request for this kind of capability. Would you want to take a stab at a `VectorDiracKernel`? Otherwise you can convert this to an issue (well we should do that anyway) and we can try to tackle it when someone has time!

**Reply by maxnezdyur** [reply]
I can take a crack at it. I'm assuming modeling it like ADKernel, templating it for VectorVariable?

**Reply by lindsayad** [reply]
That would be the most ideal
